### PR TITLE
Invoke: `docker.compilebuildtool` command

### PR DESCRIPTION
**Pull request by humitos**
Simplify how we pre-compile `build.tools` on development.

### CODE REVIEW

File: dockerfiles/tasks.py [hunk 1]
```diff
@@ -198,3 +198,75 @@ def translations(c, action):
...
+    'tool': 'build.tool to compile (python, nodejs, rust, golang)',
+    'version': 'specific version for the tool',
+    'os': 'ubuntu-20.04 or ubuntu-22.04 (default)',
+})
+def compilebuildtool(c, tool, version, os='ubuntu-22.04'):
+    """Compile a ``build.tools`` to be able to use that tool/version from a build in a quick way."""
+    DOCKER_DEFAULT_IMAGE = 'readthedocs/build'
+    # Copy&paste from ``readthedocs/settings/base.py``
+    # We cannot import it directly because it has a bunch of depedencies that are not installed.
+    # I tried using ``ast``, but it was too complex
```
[the remark this answers is not in the document]
Comment (humitos) [reply dockerfiles/tasks.py]: I refactored that setting to be on its own file `readthedocs.builds._docker`

File: dockerfiles/tasks.py [hunk 2]
```diff
@@ -198,3 +198,32 @@ def translations(c, action):
...
+@task(help={
+    'tool': 'build.tool to compile (python, nodejs, rust, golang)',
+    'version': 'specific version for the tool',
+    'os': 'ubuntu-20.04 or ubuntu-22.04 (default)',
+})
+def compilebuildtool(c, tool, version, os='ubuntu-22.04'):
+    """Compile a ``build.tools`` to be able to use that tool/version from a build in a quick way."""
+    from readthedocs.builds._docker import RTD_DOCKER_BUILD_SETTINGS
+
+    oss = RTD_DOCKER_BUILD_SETTINGS['os'].keys()
```
Comment (stsewd) [inline dockerfiles/tasks.py:212]:
```suggestion
    valid_oss = RTD_DOCKER_BUILD_SETTINGS['os'].keys()
```

File: dockerfiles/tasks.py [hunk 3]
```diff
@@ -198,3 +198,32 @@ def translations(c, action):
...
+    if os not in oss:
+        print(f'Invalid os. You must specify one of {", ".join(oss)}')
+        sys.exit(1)
+
+    tools = RTD_DOCKER_BUILD_SETTINGS['tools'].keys()
```
Comment (stsewd) [inline dockerfiles/tasks.py:217]:
```suggestion
    valid_tools = RTD_DOCKER_BUILD_SETTINGS['tools'].keys()
```

File: dockerfiles/tasks.py [hunk 4]
```diff
@@ -198,3 +198,32 @@ def translations(c, action):
...
+    'version': 'specific version for the tool',
+    'os': 'ubuntu-20.04 or ubuntu-22.04 (default)',
+})
+def compilebuildtool(c, tool, version, os='ubuntu-22.04'):
+    """Compile a ``build.tools`` to be able to use that tool/version from a build in a quick way."""
+    from readthedocs.builds._docker import RTD_DOCKER_BUILD_SETTINGS
+
+    oss = RTD_DOCKER_BUILD_SETTINGS['os'].keys()
+    if os not in oss:
+        print(f'Invalid os. You must specify one of {", ".join(oss)}')
```
Comment (stsewd) [inline dockerfiles/tasks.py:214]:
```suggestion
        print(f'Invalid OS. You must specify one of {", ".join(oss)}')
```

File: dockerfiles/tasks.py [hunk 5]
```diff
@@ -198,3 +198,32 @@ def translations(c, action):
...
+        sys.exit(1)
+
+    tools = RTD_DOCKER_BUILD_SETTINGS['tools'].keys()
+    if tool not in tools:
+        print(f'Invalid tool. You must specify one of {", ".join(tools)}')
+        sys.exit(1)
+
+    versions = RTD_DOCKER_BUILD_SETTINGS['tools'][tool].keys()
```
Comment (stsewd) [inline dockerfiles/tasks.py:222]:
```suggestion
    valid_versions = RTD_DOCKER_BUILD_SETTINGS['tools'][tool].keys()
```

**Pull request merged**